Rollbar/Rollbar.m

```diff
@@ -91,10 +91,18 @@ + (void)logWithLevel:(NSString*)level message:(NSString*)message data:(NSDiction
     [notifier log:level message:message exception:nil data:data context:context];
 }
 
++ (void)logWithLevel:(NSString*)level message:(NSString*)message exception:(NSException*)exception data:(NSDictionary*)data context:(NSString*)context {
```

Comment:
So the generic question is whether this is an interface we are happy with
```
<level>With<Type>:(<Type*>)<type> <OtherType>:(<OtherType*>)<otherType>
```

i.e. 

```
debugWithMessage:(NSString*)message data:(NSDictionary*)data
```

That is currently what we have, so adding exceptions how you have on top of that makes sense.

But I think the underlying notifier interface is more user friendly and we could hoist that up, the only hard part is trying to disambiguate level as a string from message. I think it might be better to expose an enum for level so we have a different type

```objc
typedef enum { ... } RollbarLevel;

+ (void)log:(RollbarLevel)level message:(NSString*)message exception:(NSException*)exception data:(NSDictionary*)data context:(NSString*)context
+ (void)log:(NSString*)message exception:(NSException*)exception data:(NSDictionary*)data context:(NSString*)context
+ (void)log:(RollbarLevel)level exception:(NSException*)exception data:(NSDictionary*)data context:(NSString*)context
```
And for the other convenience methods you don't have to worry about level because it is encoded in the method name, i.e.:

```objc
+ (void)debug:(NSString *)message data:(NSDictionary*)data
+ (void)debug:(NSException *)exception data:(NSDictionary*)data
```

Reply:
Changed to use enum and the new signature.